core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -740,6 +743,7 @@ private[spark] class TaskSetManager(
     }
     removeRunningTask(tid)
     info.markFinished(state)
+    successfulTaskDurations.remove(taskInfos(tid).duration)
```

Comment:
I wouldn't bother with this, because the remove method is complicated (so you can eliminate it without this line of code), and the median is good because it's not very sensitive to outlier values, so I think it's fine to leave failed tasks in.  Also, I'm not sure this code is correct, because it's possible that the duration hadn't yet been added to successfulTaskDurations, in which case you could be removing the duration that happens to be the same, but actually corresponds to a successful task.